cfme/markers/env_markers/provider.py

```diff
@@ -153,10 +153,28 @@ def add_prov(prov):
     else:
         allowed_providers = potential_providers
 
+    need_prov_keys = False
+    for filter in filters:
+        if isinstance(filter, ProviderFilter) and filter.classes:
+            for filt in filter.classes:
```

Comment:
`need_prov_keys = any(getattr(filt, 'type_name', None) is not None for fit in filter.classes)`